_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -63,276 +63,329 @@ typedef struct {
     /* CML GT2 */
-    { 0x9b41, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bca, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bcb, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bcc, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bc0, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bc5, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bc8, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bc4, MFX_HW_CFL, MFX_GT2 },
-    { 0x9bc2, MFX_HW_CFL, MFX_GT2 },
+    { 0x9B41, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BCA, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BCB, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BCC, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC0, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC5, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC8, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC4, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC2, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BC6, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BE6, MFX_HW_CFL, MFX_GT2 },
+    { 0x9BF6, MFX_HW_CFL, MFX_GT2 },
+
+    /* CFL S */
+    { 0x3E90, MFX_HW_CFL, MFX_GT1 }, /* SRV GT1 */
+    { 0x3E93, MFX_HW_CFL, MFX_GT1 }, /* SRV GT1 */
+    { 0x3E99, MFX_HW_CFL, MFX_GT1 }, /* SRV GT1 */
+
+    { 0x3E91, MFX_HW_CFL, MFX_GT2 }, /* SRV GT2 */
+    { 0x3E92, MFX_HW_CFL, MFX_GT2 }, /* SRV GT2 */
+    { 0x3E96, MFX_HW_CFL, MFX_GT2 }, /* SRV GT2 */
+    { 0x3E98, MFX_HW_CFL, MFX_GT2 }, /* SRV GT2 */
+    { 0x3E9A, MFX_HW_CFL, MFX_GT2 }, /* SRV GT2 */
+
+    /* CFL H */
+    { 0x3E9C, MFX_HW_CFL, MFX_GT2 },
+
+    { 0x3E9B, MFX_HW_CFL, MFX_GT2 }, /* Halo GT2 */
+    { 0x3E94, MFX_HW_CFL, MFX_GT2 }, /* Halo GT2 */
 
+    { 0x3EA9, MFX_HW_CFL, MFX_GT2 },
+
+    { 0x3EA5, MFX_HW_CFL, MFX_GT3 }, /* ULT GT3 */
+    { 0x3EA6, MFX_HW_CFL, MFX_GT3 }, /* ULT GT3 */
+    { 0x3EA7, MFX_HW_CFL, MFX_GT3 }, /* ULT GT3 */
+    { 0x3EA8, MFX_HW_CFL, MFX_GT3 }, /* ULT GT3 */
+
+    { 0x3EA1, MFX_HW_CFL, MFX_GT1 },
+    { 0x3EA4, MFX_HW_CFL, MFX_GT1 },
+
+    { 0x3EA0, MFX_HW_CFL, MFX_GT2 },
+    { 0x3EA3, MFX_HW_CFL, MFX_GT2 },
+
+    { 0x3EA2, MFX_HW_CFL, MFX_GT3 },
 
     /* CNL */
+    { 0x5A54, MFX_HW_CNL, MFX_GT1 },
+    { 0x5A5C, MFX_HW_CNL, MFX_GT1 },
+    { 0x5A44, MFX_HW_CNL, MFX_GT1 },
+    { 0x5A4C, MFX_HW_CNL, MFX_GT1 },
+
     { 0x5A51, MFX_HW_CNL, MFX_GT2 },
+    { 0x5A59, MFX_HW_CNL, MFX_GT2 },
+    { 0x5A41, MFX_HW_CNL, MFX_GT2 },
+    { 0x5A49, MFX_HW_CNL, MFX_GT2 },
     { 0x5A52, MFX_HW_CNL, MFX_GT2 },
     { 0x5A5A, MFX_HW_CNL, MFX_GT2 },
     { 0x5A42, MFX_HW_CNL, MFX_GT2 },
     { 0x5A4A, MFX_HW_CNL, MFX_GT2 },
-    { 0x5A59, MFX_HW_CNL, MFX_GT2 },
-    { 0x5A41, MFX_HW_CNL, MFX_GT2 },
-    { 0x5A49, MFX_HW_CNL, MFX_GT2 },
+    { 0x5A50, MFX_HW_CNL, MFX_GT2 },
+    { 0x5A40, MFX_HW_CNL, MFX_GT2 },
 
     /* ICL LP */
-    { 0xFF05, MFX_HW_ICL_LP, MFX_GT1 },
     { 0x8A50, MFX_HW_ICL_LP, MFX_GT2 },
-    { 0x8A51, MFX_HW_ICL_LP, MFX_GT2 },
-    { 0x8A52, MFX_HW_ICL_LP, MFX_GT2 },
-    { 0x8A53, MFX_HW_ICL_LP, MFX_GT2 },
-    { 0x8A56, MFX_HW_ICL_LP, MFX_GT1 },
-    { 0x8A57, MFX_HW_ICL_LP, MFX_GT1 },
-    { 0x8A58, MFX_HW_ICL_LP, MFX_GT1 },
+    { 0x8A5C, MFX_HW_ICL_LP, MFX_GT1 },
     { 0x8A59, MFX_HW_ICL_LP, MFX_GT1 },
+    { 0x8A58, MFX_HW_ICL_LP, MFX_GT1 },
```

Comment:
what's the sense to reorder them? To me it is much easier when they are ordered

Reply:
> what's the sense to reorder them?

To be able to compare with i915_pciids.h in bcompare

> To me it is much easier when they are ordered

Much easier to do what?

Reply:
> Much easier to do what?

To understand what you have added! or dropped.

I had in mind a little different approach to this change. I hoped to just copy kernel header as is and reuse.

Reply:
If you want to review, you'll have to spend some time on it..

Reply:
I am spending it already. However, you are heavily changing the code and you don't explain anywhere what you did and how others are supposed to support the code. Again, you reorder device ID list. How someone is supposed to support that? Why someone won't submit PR to reorder them back. You order is unnatural! At least provide README.md and explain what you did and how someone needs to support that/compare/etc.

Reply:
I explained it. Please see above: "To be able to compare with i915_pciids.h in bcompare"
There is a comment in code that "The list should be sync'ed manually with include/drm/i915_pciids.h". You're saying like current order is natural. No, it is not. New one is at least is synced with some reference point.

Reply:
> I explained it. Please see above: 
This should be in code base, not in a github comments.

Reply:
NP. Added. Please review,